### PR TITLE
(MODULES-560) Add new functions validate_numeric() and validate_integer().

### PR DESCRIPTION
**Pull request by poikilotherm**
I saw MODULES-560 on Jira (https://tickets.puppetlabs.com/browse/MODULES-560) and had a need for it (check ports easily in a manifest), so I implemented it.

Added rspec unit tests, too.
Added necessary stuff to README.markdown, too.

Would be glad, if this gets merged... :) Thanks!

### CODE REVIEW

**Comment by cyberious** [comment]
With the future parse and types in the next version, what is this accomplishing that you can't with validate_re and a regex like '^\-?\d+\.?*\d*$'
Just trying to understand use case.

**Comment by hlindberg** [comment]
With the new type system in Puppet 4.0 (and in future parser in 3.x) there is support for checking not only type but also the values of types. As an example, for integers (and floats) the type can be expressed as a range and the type can be combined with Optional (if an undef value is acceptable). Further you can use the scanf function to convert string representations to values (and handle failure).  Types can be combined to express things like - Array of integers in a range etc. Type checking is automatic if parameter are declared with type, and there is an assert_type function that can be used for manual type checking.

IMO we already have far too many "is_xxx" and "validate_xxx" functions in standard lib and we are about to deprecate and update these functions for puppet 4. Suggest sharing this validate_numeric function in a separate module for older puppet versions.

**Comment by poikilotherm** [comment]
Right, didn't have the new Puppet 4 stuff on the radar, but wanted to be able to do things now, still using Puppet 3.x

Is there already such a module like you suggested? This seems to be a good plan...

**Comment by hlindberg** [comment]
There is no such module (collection of 3x, new, but soon to be deprecated functions) - suggest you create your own.

File: README.markdown [hunk 1]
```diff
@@ -579,6 +579,76 @@ The first argument of this function should be the string to test, and the second
...
+
+  The following values will pass:
+
+  ```
+  validate_integer(1)
+  validate_integer(1, 2)
+  validate_integer(1, 1)
+  validate_integer(1, 2, 0)
+  validate_integer(2, 2, 2)
+  validate_integer(2, '', 0)
```
Comment (daenney) [inline README.markdown]: How does `''` get converted to an integer?

File: README.markdown [hunk 2]
```diff
@@ -579,6 +579,76 @@ The first argument of this function should be the string to test, and the second
...
+  validate_integer(2, undef, 0)
```
Comment (daenney) [inline README.markdown]: Same with `undef`?

**Comment by hlindberg** [comment]
I really object to this having been added - a function called validate_integer() that accepts undef and empty string - really? And it accepts an Array. An array is not an integer.

@Hunner I strongly suggest reverting this. It will just trick people and make them write buggy code.

**Comment by hlindberg** [comment]
I had a knee-jerk reaction and posted something that was not right. I deleted that comment. Sorry about the noise, should have read more carefully.

**Comment by poikilotherm** [comment]
Ego te absolvum.

**Pull request merged**